tests/Doctrine/Tests/ORM/Query/ExprTest.php

```diff
@@ -445,4 +445,13 @@ public function testAddNull() {
         
         $this->assertEquals(0, $andExpr->count());
     }
+
+    public function testComparisonBooleans()
+    {
+        $comparison = new Expr\Comparison('foo',Expr\Comparison::EQ, true);
+        $this->assertSame('foo = 1', (string)$comparison);
```

Comment:
isn't `__toString()` supposed to return DQL? Because the EBNF shows `true` and `false` are `true and `false` in DQL: https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/dql-doctrine-query-language.html#terminals

Reply:
I don't know the internals enough to say. All I can say is that I have the repo query as stated above, but if I pass false to the set method then I get a QueryException as follows:

```
        "class": "Doctrine\\ORM\\Query\\QueryException",
        "message": "UPDATE NS\\MidwiferyBundle\\Entity\\PotentialClient w SET w.duplicate =  WHERE w.id = :wId",
```

This test duplicates that output. and seems to mirror the issue reported in #1638.